### DISCUSSION TITLE
Doesn't it support bf16 when using Deepspeed?

### DISCUSSION THREAD

**Discussion by toriving** [Lightning-AI/pytorch-lightning#12359] [DDP / multi-GPU / multi-node] [answered]
Hello.

I am currently trying to train a model using Deepspeed zero-2.
In Deepspeed official repository, can use bf16 when using zero-2, but isn't it currently supported in PL?

**Comment by rohitgr7** [comment, marked as answer]
if it's supported natively in deepspeed, PL should support it. Can you share some references from deepspeed?

**Reply by toriving** [reply]
I was able to find a reference at the links below.
- https://github.com/microsoft/DeepSpeed/pull/1398
- https://www.deepspeed.ai/docs/config-json/#bfloat16-training-options

**Reply by rohitgr7** [reply]
PR: [Add bfloat16 support to DeepSpeedStrategy](https://github.com/PyTorchLightning/pytorch-lightning/pull/12508#)

**Reply by toriving** [reply]
Cooool! Thx :)
